src/oauth2/client.cr

```diff
@@ -145,17 +152,25 @@ class OAuth2::Client
   end
 
   private def get_access_token : AccessToken
-    body = HTTP::Params.build do |form|
-      form.add("client_id", @client_id)
-      form.add("client_secret", @client_secret)
-      yield form
-    end
-
     headers = HTTP::Headers{
       "Accept"       => "application/json",
       "Content-Type" => "application/x-www-form-urlencoded",
     }
 
+    body = HTTP::Params.build do |form|
+      case @auth_scheme
+      when .http_basic?
+        headers.add(
+          "Authorization",
+          "Basic #{Base64.strict_encode "#{@client_id}:#{@client_secret}"}"
```

Comment:
What do you think of making use of`Http::Client#basic_auth` here? I could see that it makes the rest of the code too verbose, especially the then more complex handling of `token_uri`

Reply:
Done